src/Connector.js

```diff
@@ -252,20 +261,22 @@ class Connector {
     this.onReconnectedCb = _.noop();
 
     await createService(this.iotAgentUrl, this.orionUrl, '/device', 'default', 'device');
+    await this.connectMQTT();
+    const devices = await this.listDevices();
+    await subscribeToEntities(this.client, devices);
+    this.client.on('message', async (topic, payload) => {
+      await this.messageHandler(topic, payload);
+    });
+  }
 
-    return new Promise((resolve, reject) => {
-      this.client = mqtt.connect(this.iotAgentMQTT);
+  async connectMQTT() {
+    this.client = await this.createConnection();
+  }
 
-      this.client.on('connect', async () => {
-        const devices = await this.listDevices();
-        await subscribeToEntities(this.client, devices);
-        this.client.on('message', async (topic, payload) => {
-          await this.messageHandler(topic, payload);
-        });
-        return resolve();
-      });
-      this.client.on('error', error => reject(error));
-    });
+  async createConnection() {
+    const client = mqtt.connect(this.iotAgentMQTT);
+    await promisify(client, 'connect', mqtt.connect.bind(mqtt), this.iotAgentMQTT);
+    return client;
```

Comment:
`const client = await mqtt.connectAsync(this.iotAgentMQTT);`
This promise will be resolved after receive a `connect` event

Reply:
Nops

Reply:
My bad, that was supposed to stay in draft!! :sweat: 
Made the changes requested this time, promise (haha, get it? :wink: )